src/index.js

```diff
@@ -26,12 +26,20 @@ function syncEvent(node, eventName, newEventHandler) {
   }
 }
 
-export default function (CustomElement, opts) {
+export default function (CustomElementOrTagName, opts) {
   opts = assign({}, defaults, opts);
-  if (typeof CustomElement !== 'function') {
-    throw new Error('Given element is not a valid constructor');
+  if (typeof CustomElementOrTagName !== 'function' && typeof CustomElementOrTagName !== 'string') {
+    throw new Error('Given element is not a valid constructor or tag name');
   }
-  const tagName = (new CustomElement()).tagName;
+
+  let CustomElementConstructor = CustomElementOrTagName
+  if (typeof CustomElementOrTagName === 'string') {
+    CustomElementConstructor = document.createElement(CustomElementOrTagName).constructor
```

Comment:
You could use `customElements.get(CustomElementOrTagName)`. This would have the added benefit of solving https://github.com/webcomponents/react-integration/pull/60/files#r95096012, I think.